test/robustness/overlay/areal_areal/random_integer_grids.cpp

```diff
@@ -0,0 +1,267 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Robustness Test
+
+// Copyright (c) 2025 Tinko Bartels, Shenzhen, China.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#define BOOST_GEOMETRY_NO_BOOST_TEST
+
+#include <bitset>
+#include <iostream>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include <boost/program_options.hpp>
+
+#include <boost/geometry/algorithms/difference.hpp>
+#include <boost/geometry/algorithms/union.hpp>
+#include <boost/geometry/algorithms/sym_difference.hpp>
+#include <boost/geometry/algorithms/intersection.hpp>
+#include <boost/geometry/algorithms/is_valid.hpp>
+#include <boost/geometry/io/wkt/write.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+
+#define BOOST_GEOMETRY_TEST_ONLY_ONE_TYPE
+#define BOOST_GEOMETRY_DEFAULT_TEST_TYPE int
+#include <geometry_test_common.hpp>
+
+constexpr int chunk_size = 64;
+using bits = std::vector<std::bitset<chunk_size>>;
+using prng = std::mt19937_64;
+static_assert(sizeof(prng::result_type) >= chunk_size / 8, "PRNG output too narrow.");
+
+namespace bg = boost::geometry;
+
+using point = bg::model::d2::point_xy<int>;
+using box   = bg::model::box<point>;
+using mp    = bg::model::multi_polygon<bg::model::polygon<point>>;
+
+struct grid_settings
+{
+    int width              = 5;
+    int height             = 5;
+    int count              = 1;
+    prng::result_type seed = prng::default_seed;
+};
+
+constexpr int cell_dimension = 2;
+
+std::vector<box> grid_cells(grid_settings const& settings)
+{
+    std::vector<box> out;
+    for (int y = 0; y < settings.height; ++y)
+    {
+        for (int x = 0; x < settings.width; ++x)
+        {
+            out.push_back(box{point{x * cell_dimension, y * cell_dimension},
+                              point{(x + 1) * cell_dimension, (y + 1) * cell_dimension}});
+        }
+    }
+    return out;
+}
+
+std::vector<point> test_points(grid_settings const& settings)
+{
+    std::vector<point> out;
+    for (int y = 0; y < settings.height; ++y)
+    {
+        for (int x = 0; x < settings.width; ++x)
+        {
+            out.push_back(point{x * cell_dimension + cell_dimension / 2,
+                                y * cell_dimension + cell_dimension / 2});
+        }
+    }
+    return out;
+}
+
+std::ostream& operator<<(std::ostream& os, std::pair<bits, grid_settings> const& b_gs)
+{
+    os << '\n';
+    for (int y = b_gs.second.height - 1; y >= 0; --y)
+    {
+        for (int x = 0; x < b_gs.second.width; ++x)
+        {
+            int index = y * b_gs.second.width + x;
+            os << b_gs.first[index / chunk_size][index % chunk_size];
+        }
+        os << '\n';
+    }
+    return os;
+}
+
+bits geometry_to_bits(mp const& g, std::vector<point> const& tp)
```

Comment:
The names are short and sometimes cryptic.
I don't say I never use abbreviations myself (I do). But these ones I don't fully get all.
`tp`, what is it?
`b` is probably fine in this context.
`g` is short (but at least I understand it), I would use `geometry` here or make the `mp` a `multi_polygon_t` or `mp_t` and call it `mp` then here
`bm` (below)
`prng` and `gen`, I would use longer names